packages/core/src/controllers/NetworkController.ts

```diff
@@ -114,6 +125,12 @@ export const NetworkController = {
     }
   },
 
+  checkIfSmartAccountEnabled() {
+    const networkId = Number(state.caipNetwork?.id?.split(':')?.[1])
```

Comment:
I'm not really sure that we should have a util function about getting the network id from the caip id since we are doing the same parsing in different places. A util method like; 

```js
function parseCAIPNetwork(caipNetwork) { ... }
..
const { networkId } = parseCAIPNetwork("..")
```

would be nice maybe?

Reply:
agree, will implement

Reply:
moved caipNetworkIdToNumber to core helper utils since it needs to be used outside scaffold. 
we need to review what methods should be in what files for the utils as they are currently scattered around the app.
that's scoped for tech debt week work